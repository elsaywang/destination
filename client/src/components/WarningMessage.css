:local(.warningMessage) {
    display: flex;
    margin-left: 0.5em;
    padding: 0.15em;
    color: var(--spectrum-semantic-notice-color-status);
    justify-content: flex-end;
}
:local(.messageBody) {
    margin-left: 0.5em;
    font-size: var(--spectrum-global-font-size-medium);
    font-weight: var(--spectrum-global-font-weight-regular);
    font-style: italic;
}
