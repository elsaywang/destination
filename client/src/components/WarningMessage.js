import React from 'react';
import PropTypes from 'prop-types';
import styles from './WarningMessage.css';
import Alert from '@react/react-spectrum/Icon/Alert';

const SignalsSelectionWarningMessageBody = () => (
    <span className={styles.messageBody}>
        We canâ€™t create traits from signals of different types. Try again using signals all from
        either <b> Real-Time Signals </b> or <b> Onboarded Signals </b>.
    </span>
);

export const messageBodyTemplateMapping = {
    signalsSelection: <SignalsSelectionWarningMessageBody />,
};

export const WarningMessage = ({ warningType }) => {
    return (
        <div className={styles.warningMessage}>
            <Alert size="S" variant="warning" />
            {messageBodyTemplateMapping[warningType]}
        </div>
    );
};

WarningMessage.propTypes = {
    warningType: PropTypes.string,
};
export default WarningMessage;
