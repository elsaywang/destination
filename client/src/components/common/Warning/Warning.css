:local(.warning) {
    display: flex;
    padding-bottom: 1em;
    color: var(--spectrum-semantic-notice-color-status);
    justify-content: flex-end;
}
:local(.warningMessage) {
    margin-left: 0.5em;
    font-size: var(--spectrum-global-font-size-medium);
    font-weight: var(--spectrum-global-font-weight-regular);
    font-style: italic;
}
