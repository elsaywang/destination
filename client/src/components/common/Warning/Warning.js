import React from 'react';
import styles from './Warning.css';
import Alert from '@react/react-spectrum/Icon/Alert';

const Warning = props => {
    return (
        <div className={styles.warning}>
            <Alert size="S" variant="warning" />
            <div className={styles.warningMessage}>{props.children}</div>
        </div>
    );
};

export default Warning;
