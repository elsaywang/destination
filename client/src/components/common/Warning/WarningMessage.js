import React from 'react';
import PropTypes from 'prop-types';

const WarningMessage = ({ content }) => <span>{content}</span>;

WarningMessage.propTypes = {
    content: PropTypes.string,
};
export default WarningMessage;
