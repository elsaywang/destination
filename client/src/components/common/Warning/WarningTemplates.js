import React from 'react';
import PropTypes from 'prop-types';

export const SignalsSelectionWarningMessage = () => (
    <span>
        We canâ€™t create traits from signals of different types. Try again using signals all from
        either <strong> Real-Time Signals </strong> or <strong> Onboarded Signals</strong>.
    </span>
);

export const SignalsSelectionLimitMessage = ({ maxSignalSelections }) => (
    <span>
        You may select up to <strong>{maxSignalSelections}</strong> Signals. Only the first{' '}
        <strong>{maxSignalSelections}</strong> selected signals will be used to create a trait.
    </span>
);

SignalsSelectionLimitMessage.propTypes = {
    maxSignalSelections: PropTypes.number.isRequired,
};
