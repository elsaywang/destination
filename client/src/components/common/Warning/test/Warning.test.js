import React from 'react';
import { shallow } from 'enzyme';
import Warning from '../Warning';
import WarningMessage from '../WarningMessage';
import { SignalsSelectionWarningMessage } from '../WarningTemplates';
import Alert from '@react/react-spectrum/Icon/Alert';

describe('<Warning/> component', () => {
    describe('rendering when it does not contain any children component', () => {
        const wrapper = shallow(<Warning />);

        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders `warning` <Alert/> icon with size=`S`', () => {
            expect(wrapper.find(Alert).exists()).toBeTruthy();
            expect(wrapper.find(Alert).props().size).toEqual('S');
            expect(wrapper.find(Alert).props().variant).toEqual('warning');
        });
    });

    describe('rendering when it includes <WarningMessage/> as children component', () => {
        const { content } = 'there is a warning';
        const wrapper = shallow(
            <Warning>
                <WarningMessage {...content} />
            </Warning>,
        );

        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders <WarningMessage /> component with correct `content` props', () => {
            expect(wrapper.find(WarningMessage).exists()).toBeTruthy();
            expect(wrapper.find(WarningMessage).props().content).toEqual(content);
        });
    });

    describe('rendering when it includes <SignalsSelectionWarningMessage/> as children component', () => {
        const { content } = 'there is a warning';
        const wrapper = shallow(
            <Warning>
                <SignalsSelectionWarningMessage />
            </Warning>,
        );

        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders <SignalsSelectionWarningMessage/> component ', () => {
            expect(wrapper.find(SignalsSelectionWarningMessage).exists()).toBeTruthy();
        });
    });
});
