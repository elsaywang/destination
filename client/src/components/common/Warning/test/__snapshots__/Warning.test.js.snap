// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Warning/> component rendering when it does not contain any children component matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Warning />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Alert
          size="S"
          variant="warning"
/>,
        <div
          className={undefined}
/>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "size": "S",
          "variant": "warning",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Alert
            size="S"
            variant="warning"
/>,
          <div
            className={undefined}
/>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": "S",
            "variant": "warning",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Warning/> component rendering when it includes <SignalsSelectionWarningMessage/> as children component matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Warning>
    <Unknown />
</Warning>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Alert
          size="S"
          variant="warning"
/>,
        <div
          className={undefined}
>
          <Unknown />
</div>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "size": "S",
          "variant": "warning",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Unknown />,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Alert
            size="S"
            variant="warning"
/>,
          <div
            className={undefined}
>
            <Unknown />
</div>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": "S",
            "variant": "warning",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Unknown />,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Warning/> component rendering when it includes <WarningMessage/> as children component matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Warning>
    <WarningMessage />
</Warning>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Alert
          size="S"
          variant="warning"
/>,
        <div
          className={undefined}
>
          <WarningMessage />
</div>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "size": "S",
          "variant": "warning",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WarningMessage />,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Alert
            size="S"
            variant="warning"
/>,
          <div
            className={undefined}
>
            <WarningMessage />
</div>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": "S",
            "variant": "warning",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WarningMessage />,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
